### ISSUE TITLE
Compiling fails on OpenBSD

### ISSUE THREAD

**Issue by Midar** [bebbo/amiga-gcc#115]
After a few minor adjustments (change /bin/bash in Makefile to just bash, add the same flags to GCC as on macOS, using clang instead of gcc, disabling gdb), GCC compiles, but compiling newlib fails with:

```
m68k-amigaos-gcc  -fbaserel -I/home/js/Devel/amiga-gcc/build-OpenBSD/newlib/libb/newlib/targ-include -I/home/js/Devel/amiga-gcc/projects/newlib-cygwin/newlib/libc/include -DPACKAGE_NAME=\"newlib\" -DPACKAGE_TARNAME=\"newlib\" -DPACKAGE_VERSION=\"3.0.0\" -DPACKAGE_STRING=\"newlib\ 3.0.0\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -I. -I/home/js/Devel/amiga-gcc/projects/newlib-cygwin/newlib/libc/argz -DCOMPACT_CTYPE -DSIGNAL_PROVIDED -fno-builtin      -O2 -fomit-frame-pointer  -fbaserel -c -o lib_a-dummy.o `test -f 'dummy.c' || echo '/home/js/Devel/amiga-gcc/projects/newlib-cygwin/newlib/libc/argz/'`dummy.c
cc1: fatal error: PACKAGE_NAME="newlib": No such file or directory
compilation terminated.
```

In fact, compiling any file with -D fails:

```
$ m68k-amigaos-gcc -DASD test.c
cc1: fatal error: ASD: No such file or directory
compilation terminated.
```

So it seems like this is not passed correctly to cc1.

One thing I also noticed is that GCC tries to use -no-pie, which is unavailable on OpenBSD (similarly to macOS, all binaries are PIE).

**Comment by bebbo** [comment]
it builds on that wanky shanker OS - see https://github.com/bebbo/amiga-gcc/blob/master/.travis.yml

**Comment by Midar** [comment]
I don't see anything OpenBSD in there, and AFAIK Travis also don't supports OpenBSD?

I'm trying this on a regular OpenBSD/amd64 6.5, fresh install.

**Comment by bebbo** [comment]
> 
> 
> I don't see anything OpenBSD in there, and AFAIK Travis also don't supports OpenBSD?
> 
> I'm trying this on a regular OpenBSD/amd64 6.5, fresh install.

lol - sorry  - my fault

**Comment by Midar** [comment]
The following ktrace looks like cc1 is called correctly:

```
 22039 m68k-amigaos-gcc ARGS  
        [0] = "/opt/amiga/libexec/gcc/m68k-amigaos/6.5.0b/cc1"
        [1] = "-quiet"
        [2] = "-isystem"
        [3] = "-D"
        [4] = "FOO"
        [5] = "test.c"
        [6] = "-quiet"
        [7] = "-dumpbase"
        [8] = "test.c"
        [9] = "-mcpu=68000"
        [10] = "-auxbase"
        [11] = "test"
        [12] = "-o"
        [13] = "/tmp//cced0vNT.s"
 22039 cc1      NAMI  "/usr/libexec/ld.so"
 22039 cc1      RET   execve 0
```

**Comment by Midar** [comment]
Sorry, I didn't look carefully enough: The -isystem before the -D makes it take the -D as an argument to -isystem, causing the FOO to become an input file.

**Comment by bebbo** [comment]
if `-isystem` is empty some property might be missing

**Comment by bebbo** [comment]
the include path - maybe no valid path is available?

**Comment by Midar** [comment]
I guess it's trying to add -isystem to add the libnix headers, but for whatever reason not finding them? Do you have any idea where this could happen?

**Comment by bebbo** [comment]
it should read `-isystem /opt/amiga/include`

**Comment by Midar** [comment]
Where is this /opt/amiga/include usually coming from? I'm guessing the PREFIX is passed on somewhere, but this fails. Possibly due to some tool not being a GNU tool but a BSD tool or something? awk is a classic for these things.

**Comment by Midar** [comment]
Removing all the -isystem lines from gcc/config/m68k/m68kamigaos.h seems to make it work. I guess the %:sdk_root() function (or whatever it is) does not work properly and evaluates to empty.

**Comment by bebbo** [comment]
this means that `/proc/self/exe` is not available to determine the own executables path.

Is there an alternative?

**Comment by Midar** [comment]
Yes, /proc/self/exe is a Linux-only thing. I'm currently looking at amiga_m68k_prefix_func(), which seems where sdk_root is coming from, but it doesn't seem to be using /proc/self/exe - that would also fail on Windows and macOS, I suppose?

I suppose a portable way would be to look at argv[0]: If it's absolute, you're done, and if it's relative, you look for argv[0] in PATH.

**Comment by bebbo** [comment]
feel free to improve `/gcc/libiberty/make-relative-prefix.c`

**Comment by Midar** [comment]
There seems to be no good way to do this on OpenBSD. How about making it detect the path at runtime for Windows / mac / Linux like it is now, and for everything else use the prefix used at compile time? Seems like the easiest solution?

**Comment by Midar** [comment]
Using the prefix specified at compile time here makes it work on OpenBSD. At least, it built libnix now :)

**Comment by Midar** [comment]
Ok, I was too fast: libgcc does not build:

```
checking for shl_load... configure: error: Link tests are not allowed after GCC_NO_EXECUTABLES.
gmake[1]: *** [Makefile:10214: configure-target-libstdc++-v3] Error 1
```

When trying to set the ac_cv_* vars, it will just fail at the next one, so I guess something more generic is wrong.

**Comment by bebbo** [comment]
Feel free to provide a reasonable patch for OpenBSD or other systems
